awsshell/wizard.py

```diff
@@ -0,0 +1,141 @@
+        self.next_stage = spec.get('NextStage', None)
+        self.resolution = spec.get('Resolution', None)
+        self.interaction = spec.get('Interaction', None)
+
+    def __handle_static_retrieval(self):
+        return self.retrieval.get('Resource')
+
+    def __handle_request_retrieval(self):
+        # TODO very basic requests... refactor needed
+        req = self.retrieval['Resource']
+        # initialize botocore session and client for service in the request
+        session = botocore.session.get_session()
+        client = session.create_client(req['Service'])
+        # get the operation from the client
+        operation = getattr(client, xform_name(req['Operation']))
+        # get any parameters
+        parameters = req.get('Parameters', {})
+        env_parameters = self._resolve_parameters(req.get('EnvParameters', {}))
+        # union of parameters and env_parameters, conflicts favor env_params
+        parameters = dict(parameters, **env_parameters)
+        # execute operation passing all parameters
+        return operation(**parameters)
+
+    def _handle_retrieval(self):
+        print(self.prompt)
+        # In case of no retrieval, empty dict
+        if not self.retrieval:
+            return {}
+        elif self.retrieval['Type'] == 'Static':
+            data = self.__handle_static_retrieval()
+        elif self.retrieval['Type'] == 'Request':
+            data = self.__handle_request_retrieval()
+        # Apply JMESPath query if given
+        if self.retrieval.get('Path'):
+            data = jmespath.search(self.retrieval['Path'], data)
+        return data
+
+    def _resolve_parameters(self, keys):
+        for key in keys:
+            keys[key] = self._wizard_env.retrieve(keys[key])
+        return keys
+
+    def _handle_interaction(self, data):
+        # TODO actually implement this step
+        # if no interaction step, just forward data
+        if not self.interaction:
+            return data
+        elif self.interaction['ScreenType'] == 'SimpleSelect':
+            data = data[0]
+        elif self.interaction['ScreenType'] == 'SimplePrompt':
+            for field in data:
+                data[field] = 'random'
+        return data
+
+    def _handle_resolution(self, data):
+        if self.resolution:
+            if self.resolution.get('Path'):
+                data = jmespath.search(self.resolution['Path'], data)
+            self._wizard_env.store(self.resolution['Key'], data)
+
+    def get_next_stage(self):
+        if not self.next_stage:
+            return None
+        elif self.next_stage['Type'] == 'Name':
+            return self.next_stage['Name']
+        elif self.next_stage['Type'] == 'Variable':
+            return self._wizard_env.retrieve(self.next_stage['Name'])
+
+    # Executes all three steps of the stage
+    def execute(self):
+        retrieved = self._handle_retrieval()
+        transformed = self._handle_interaction(retrieved)
```

Comment:
The names retrieved and transformed are a bit opaque. Maybe something like `retrieved_options`, something that indicates that this is candidate data. For `transformed`, something like `selected_data` or something that indicates this is data that was selected by the user (or automatically).